docs_src/microservices/application/V2Migration.md

```diff
@@ -0,0 +1,298 @@
+
+!!! example "Example - Setting Functions Pipeline"
+
+    ```go
+    if err := service.SetFunctionsPipeline(
+    	transforms.NewFilterFor(deviceNames).FilterByDeviceName,
+    	transforms.NewConversion().TransformToXML,
+    	transforms.NewHTTPSender(exportUrl, "application/xml", false).HTTPPost,
+    ); err != nil {
+    	lc.Errorf("SetFunctionsPipeline returned error: %s", err.Error())
+    	os.Exit(-1)
+    }
+    ```
+
+##### MakeItRun
+
+The `MakeItRun` API has not changed.
+
+!!! example "Example - Call to MakeItRun"
+
+    ```go
+    err = service.MakeItRun()
+    if err != nil {
+    	lc.Errorf("MakeItRun returned error: %s", err.Error())
+    	os.Exit(-1)
+    }
+    ```
+
+#### Custom Pipeline Functions
+
+##### Pipeline Function signature
+
+The major change to custom Pipeline Functions for EdgeX 2.0 is the new function signature which drives all the other changes.
+
+!!! example "Example - New Pipeline Function signature"
+
+    ```go
+    type AppFunction = func(cxt AppFunctionContext, data interface{}) (bool, interface{})
+    ```
+
+This function signature passes in an instance of the new AppFunctionContext API for the context and now has only a single `data` instance for the function to operate on.
+
+##### Return Values
+
+The definitions for the Pipeline Function return values have not changed.
+
+##### Data
+
+The `data` passed in either set to a single instance for the function to process or nil.  Now you no longer need to check the length of the incoming data.
+
+!!! example - Example - Validating data before processing
+
+    ```go
+    	if data == nil {
+    		return false, errors.New("No Data Received")
+    	}
+    ```
+
+##### Logging Client
+
+The `Logging` client is now accessible from the `ctx.LoggingClient()` API. 
+
+##### Clients
+
+The available clients have changed with a few additions and `ValueDescriptorClient` has been removed. See the [Context Clients](../AppFunctionContextAPI/#clients) section for complete list of available clients.
+
+##### ResponseData
+
+The `SetResponseData` and `ResponseData` APIs replace the previous `Complete` function and direct access to the `OutputData` field.
+
+##### ResponseContentType
+
+The `SetResponseContentType` and `ResponseContentType` APIs replace the previous direct access to the `ResponseContentType` field.
+
+##### RetryData
+
+The `SetRetryData` API replaces the `SetRetryData` function and direct access to the `RetryData` field.
+
+##### MarkAsPushed
+
+The `MarkAsPushed` capability has been removed
+
+##### PushToCore
+
+The `PushToCore` API replaces the `PushToCoreData` function. The API signature has changed. See the [PushToCore](../AppFunctionContextAPI/#pushtocore) section for more details.
+
+##### New Capabilities
+
+Some new capabilities have been added to the new AppFunctionContext API. See the [App Function Context](../AppFunctionContextAPI) API section for complete details.
+
+## Custom Profiles
```

Comment:
I'd call this App Service Configurable Profiles.  It didn't strike me that this was applicable to app service config until I read the next paragraph.  In the other documentation you just call them profiles.

I'd also reference the page: https://docs.edgexfoundry.org/2.0/microservices/application/AppServiceConfigurable/#deploying-multiple-instances-using-profiles

Reply:
Done